testing/src/main/java/uk/gov/pay/commons/testing/pact/consumers/PactProviderRule.java

```diff
@@ -0,0 +1,167 @@
+        Pact pactAnnotation = method.getAnnotation(Pact.class);
+        PactDslWithProvider dslBuilder = ConsumerPactBuilder.consumer(pactAnnotation.consumer()).hasPactWith(provider);
+        RequestResponsePact pact;
+        try {
+            pact = (RequestResponsePact) method.invoke(target, dslBuilder);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to invoke pact method", e);
+        }
+        PactVerificationResult result = runPactTest(base, pact);
+        JUnitTestSupport.validateMockServerResult(result);
+    }
+
+    private Optional<PactVerification> findPactVerification(PactVerifications pactVerifications) {
+        PactVerification[] pactVerificationValues = pactVerifications.value();
+        return Arrays.stream(pactVerificationValues).filter(p -> {
+            String[] providers = p.value();
+            if (providers.length != 1) {
+                throw new IllegalArgumentException(
+                        "Each @PactVerification must specify one and only provider when using @PactVerifications");
+            }
+            String provider = providers[0];
+            return provider.equals(this.provider);
+        }).findFirst();
+    }
+
+    private Optional<Method> findPactMethod(PactVerification pactVerification) {
+        String pactFragment = pactVerification.fragment();
+        for (Method method : target.getClass().getMethods()) {
+            Pact pact = method.getAnnotation(Pact.class);
+            if (pact != null && pact.provider().equals(provider)
+                    && (pactFragment.isEmpty() || pactFragment.equals(method.getName()))) {
+
+                validatePactSignature(method);
+                return Optional.of(method);
+            }
+        }
+        return Optional.empty();
+    }
+
+    private void validatePactSignature(Method method) {
+        boolean hasValidPactSignature =
+                RequestResponsePact.class.isAssignableFrom(method.getReturnType())
+                        && method.getParameterTypes().length == 1
+                        && method.getParameterTypes()[0].isAssignableFrom(PactDslWithProvider.class);
+
+        if (!hasValidPactSignature) {
+            throw new UnsupportedOperationException("Method " + method.getName() +
+                    " does not conform required method signature 'public RequestResponsePact xxx(PactDslWithProvider builder)'");
+        }
+    }
+
+    private PactVerificationResult runPactTest(final Statement base, RequestResponsePact pact) {
+        return runConsumerTest(pact, config, mockServer -> {
+            this.mockServer = mockServer;
+            base.evaluate();
+            this.mockServer = null;
+        });
+    }
+
+    protected Map<String, RequestResponsePact> getPacts(String methodName) {
+        HashMap<String, RequestResponsePact> pacts = new HashMap<>();
+        for (Method m : target.getClass().getMethods()) {
+            if (m.getName().equals(methodName)) {
```

Comment:
we need this line because we only want to load the pact for the particular unit test. otherwise all the pacts included in the test class's methods will get loaded and expectations will fail.